### PR TITLE
fix: buy order max input and format amount to spend

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/regen-registry#1164

- add validation to credit count in `BuyCreditsModal`
- fix broken error display for `NumberTextField` component

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood` and `hambach` (see below)

#### Setting up backport PRs

After merging your PR to `master`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR, `@Mergifyio backport redwood hambach`.

2. If your branch does have merge commits:

    a. Pull latest `master`, `hambach` and `redwood` branches

    b. Create new branches for backports and merge `master` (replace `<PR#>` with your PR #)
    ```
    git checkout -b hambach-backport-<PR#> hambach
    git merge master
    git push origin hambach-backport-<PR#>
    git checkout -b redwood-backport-<PR#> redwood
    git merge master
    git push origin redwood-backport-<PR#>`
    ```

    c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.
  

### How to test

1. Go to https://deploy-preview-1186--regen-registry.netlify.app/projects
2. Click buy ecocredits button
3. Select a sell order
4. Try to input a number greater than the number of credit available in the sell order
5. Click away or submit the form

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-components/src/components/inputs/NumberTextField.tsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ const useStyles = makeStyles((theme: Theme) => ({
     position: 'absolute',
     right: 0,
     top: 0,
-    height: '100%',
+    height: 60,
```
Comment (flagrede) [inline web-components/src/components/inputs/NumberTextField.tsx:24]: Match input height.
Before we had this:
image

File: web-registry/src/components/organisms/BuyCreditsModal/BuyCreditsModal.tsx [hunk 2]
```diff
@@ -203,36 +210,53 @@ const BuyCreditsModal: React.FC<BuyCreditsModalProps> = ({
...
-                          =
-                        </Title>
-                        <div
-                          className={cx(styles.flexColumn, styles.marginRight)}
+                        <Box
+                          sx={{
+                            display: 'flex',
+                            alignItems: 'center',
+                            height: 60,
+                          }}
```
Comment (flagrede) [inline web-registry/src/components/organisms/BuyCreditsModal/BuyCreditsModal.tsx:223]: Add a wrapper with some style to keep the same alignment with and without an error message.

File: web-registry/src/components/organisms/BuyCreditsModal/BuyCreditsModal.utils.ts [hunk 3]
```diff
@@ -46,3 +47,23 @@ export const handleBuyCreditsSubmit = async (
   };
   await onSubmit(fullValues);
 };
+
+export const getCreditCountValidation =
```
Comment (flagrede) [inline web-registry/src/components/organisms/BuyCreditsModal/BuyCreditsModal.utils.ts:51]: New validation function

File: web-registry/src/components/organisms/BuyCreditsModal/BuyCreditsModal.utils.ts [hunk 4]
```diff
@@ -46,3 +47,23 @@ export const handleBuyCreditsSubmit = async (
...
+type AmountToSpendParams = {
+  creditCount: number;
+  askAmount: number;
+};
+
+export const amountToSpend = ({
+  creditCount,
+  askAmount,
+}: AmountToSpendParams): String =>
+  formatNumber(creditCount * microToDenom(askAmount), 2) ?? '-';
```
Comment (flagrede) [inline web-registry/src/components/organisms/BuyCreditsModal/BuyCreditsModal.utils.ts:69]: Trim the amount to spend decimals to 2, before we could have number like this:
image

**Comment by flagrede** [comment]
@erikalogie you can try out the fix to validate the max input in buy credit modal: https://deploy-preview-1186--regen-registry.netlify.app/projects

**Comment by erikalogie** [comment]
This looks great! Let's just shorten the text a bit to: "Must be less than or equal to the max credit(s) available (97)."

Also, can you change the line-height to 1.2 or 1.3 to tighten it up a bit?

**Review by mhagel — APPROVED** [review]
thanks!

File: web-registry/src/components/organisms/BuyCreditsModal/BuyCreditsModal.tsx [hunk 5]
```diff
@@ -203,36 +210,53 @@ const BuyCreditsModal: React.FC<BuyCreditsModalProps> = ({
...
+                                <Title variant="h4" sx={{ mr: 1.5 }}>
+                                  {amountToSpend({
+                                    askAmount: Number(
+                                      selectedSellOrder?.askAmount,
+                                    ),
+                                    creditCount: values?.creditCount,
+                                  })}
+                                </Title>
+                              </Box>
+                              <Label size="sm" sx={{ color: 'info.dark' }}>
```
Comment (haveanicedavid) [inline web-registry/src/components/organisms/BuyCreditsModal/BuyCreditsModal.tsx:254]: just FYI - you can pass `color` directly if you want (`<Label size=“sm” color=“info.dark”>`

**Review by haveanicedavid — APPROVED** [review]
lgtm

**Comment by flagrede** [comment]
@Mergifyio backport redwood hambach

**Pull request merged**